### PR TITLE
createVariable API to produce result with unitialized variance.

### PR DESCRIPTION
**Pull request by igudich**
Some preparations to refactor `MakeVariableWithVariances`.

### CODE REVIEW

File: core/include/scipp/core/variable.h [hunk 1]
```diff
@@ -557,24 +557,51 @@ makeVariableWithVariances(const Dimensions &dimensions,
...
+/// 1. If neither Values nor Variances are provided, resulting Variable contains
+/// ONLY values of corresponding length.
+/// 2. The Variances can't be provided without any Values.
+/// 3. Non empty Values and/or Variances should be consistent with shape.
+/// 4. If empty Values and/or Variances are provided, resulting Variable
+/// contains uninitialized Values and/or Variances, the only way to make
+/// Variable which contains both Values and Variances given length uninitialised
+/// is:
+///                  createVariable<T>(Dims{Dim::X}, Shape{5}, Values(0),
+///                  Variances(0)).
```
Comment (SimonHeybrock) [inline core/include/scipp/core/variable.h]: How about `Values()` (no arguments) instead of `Values(0)`?

File: core/include/scipp/core/variable.h [hunk 2]
```diff
@@ -557,24 +557,51 @@ makeVariableWithVariances(const Dimensions &dimensions,
...
-  if (val)
-    return Variable(u, dms, std::move(*val));
-  if (var)
-    throw except::VariancesError("Can't have variance without values");
-  else {
-    if constexpr (is_sparse_container<T>::value)
-      return Variable(u, dms, Vector<T>(dms.volume()));
+  if (val && var) {
+    if (val->size() == 0 && var->size() == 0)
+      return from_dimensions_and_unit<T>(dms, u);
```
Comment (SimonHeybrock) [inline core/include/scipp/core/variable.h]: Isn't it inconsistent to ignore the shape of values and variances in this special case? Also, this looks like the same call as in the case without variances below, so you are not actually making variances?

Comment (igudich) [reply core/include/scipp/core/variable.h]: The call is wrong, will fix. But the shape is not ignored: it is like if Values and Variances both empty, then construct Variable just with the corresponding Shape, if they are not, then use the normal constructor, which checks the consistency. Could you elaborate what is inconsistent?

Comment (SimonHeybrock) [reply core/include/scipp/core/variable.h]: What is does is not inconsistent, I meant it is inconsistent for the user interfce (but that is addressed in my other comment, above).

**Pull request merged**